zilliqa/src/consensus.rs

```diff
@@ -233,39 +233,26 @@ impl Consensus {
         let (start_view, finalized_view, high_qc) = {
             match db.get_high_qc()? {
                 Some(qc) => {
-                    let high_block = db
-                        .get_block_by_hash(&qc.block_hash)?
-                        .ok_or_else(|| anyhow!("missing block that high QC points to!"))?;
-                    let finalized_number = db
+                    let finalized_view = db
                         .get_finalized_view()?
                         .ok_or_else(|| anyhow!("missing latest finalized view!"))?;
                     let finalized_block = db
-                        .get_block_by_view(finalized_number)?
+                        .get_block_by_view(finalized_view)?
                         .ok_or_else(|| anyhow!("missing finalized block!"))?;
 
-                    // If latest view was written to disk then always start from there. Otherwise start from (highest out of high block and finalised block) + 1
+                    state.set_to_root(finalized_block.header.state_root_hash.into());
+
+                    // If latest view was written to disk then always start from there. Otherwise start from finalised view + 1
                     let start_view = db
                         .get_view()?
-                        .or_else(|| {
-                            Some(std::cmp::max(high_block.view(), finalized_block.view()) + 1)
-                        })
+                        .or_else(|| Some(finalized_block.view() + 1))
                         .unwrap();
 
                     trace!(
-                        "recovery: high_block view {0}, finalized_number {1}, start_view {2}",
-                        high_block.view(),
-                        finalized_number,
-                        start_view
+                        "recovery: finalized_view {0}, start_view {1}",
+                        finalized_view, start_view
                     );
 
-                    if finalized_number > high_block.view() {
-                        // We know of a finalized view higher than the view in finalized_number; start there.
-                        state.set_to_root(finalized_block.header.state_root_hash.into());
```

Comment:
Should we keep this line?

Reply:
Its sitll there I moved it up a few lines (:243)